### PR TITLE
Dialog: Add `sx` styles back to the footer styled-component & add visual regression tests for both Dialogs

### PR DESCRIPTION
**Pull request by broccolinisoup**
Caught [an issue on the dotcom integration tests](https://gh.io/AAnj9jm) where our dialog was showing the footer when it is not supposed to. ([file reference](https://gh.io/AAnimec)). It turns out the `sx` prop wasn't applied in the below code. I think https://github.com/primer/react/pull/3918 introduced a regression. I _believe_ [removing the `sx`](https://github.com/primer/react/pull/3918/files#diff-7c9e4f2540e063c93f543e6eaa44cac7dc3b0e55a8fe81030c9e571f72950a3eL384) styles from the footer styled-component wasn't intentional. 

```
<Dialog.Footer sx={{display: displayMode === DisplayMode.IssueCreation ? 'flex' : 'none'}}>
     <CreateIssueFooter onClose={() => onSafeClose()} />
</Dialog.Footer>
```

| Before | After |
|--------|--------|
|prc story that shows the dialog button has a white background, because sx styles are not applied| prc story that shows the dialog button has a yellow background, because sx styles are applied|

#### New

Added visual regression tests for Dialog and Dialog1. (We should be able to catch these issues on our end before merged now 🤗)

#### Changed

Added the `sx` styles back to the styled footer. 

### Rollout strategy

- [x] Patch release
- [ ] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan

### Testing & Reviewing

Can be looked into the custom rendering story and make sure the sx styles are applied.

### Merge checklist

- [x] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews (Storybook)
- [x] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [x] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge
- [ ] Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

**Review by dylanatsmith — APPROVED** [review]
Oof, sorry about this! Good catch.

File: e2e/components/Dialog.test.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      })
+    }
+  })
+
+  test.describe('Stress Test', () => {
+    for (const theme of themes) {
+      test.describe(theme, () => {
+        test('default @vrt', async ({page}) => {
+          await visit(page, {
+            id: 'components-dialog--features--stress-test',
```
Comment (joshblack) [inline e2e/components/Dialog.test.ts:40]:
```suggestion
            id: 'components-dialog-features--stress-test',
```

File: e2e/components/Dialog.test.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            },
+          })
+
+          await page.getByRole('button', {name: 'Show dialog'}).click()
+          expect(await page.screenshot({animations: 'disabled'})).toMatchSnapshot(`Dialog.Stress Test.${theme}.png`)
+        })
+
+        test('axe @aat', async ({page}) => {
+          await visit(page, {
+            id: 'components-dialog--features--stress-test',
```
Comment (joshblack) [inline e2e/components/Dialog.test.ts:52]:
```suggestion
            id: 'components-dialog-features--stress-test',
```

File: script/generate-e2e-tests.js [hunk 3]
```diff
@@ -444,6 +444,36 @@ const components = new Map([
...
+  [
+    'Dialog',
+    {
+      stories: [
+        {
+          id: 'components-dialog--default',
+          name: 'Default',
+        },
+        {
+          id: 'components-dialog--features--stress-test',
```
Comment (joshblack) [inline script/generate-e2e-tests.js:456]:
```suggestion
          id: 'components-dialog-features--stress-test',
```

**Review by joshblack — COMMENTED** [review]
These should help with the timeouts 🤞

**Review by pksjce — APPROVED** [review]
Love the tests! 🥳

**Pull request merged**